### DISCUSSION TITLE
Documenting Custom Commands

### DISCUSSION THREAD

**Discussion by compassoLeonardo** [cypress-io/cypress#20364] [Questions and Help]
Hello everyone, I would like to know if there is any way to document Cypress's custom commands so that their information is shown in their calls, something like a Javadoc to facilitate maintenance and development, but in JS.
Thanks!

**Comment by rmnrdi** [comment]
I believe you want JSDoc to accomplish this:
https://docs.cypress.io/guides/tooling/typescript-support#Types-for-Custom-Commands 


// cypress/support/index.ts
declare global {
  namespace Cypress {
    interface Chainable {
      /**
       * Custom command to select DOM element by data-cy attribute.
       * @example cy.dataCy('greeting')
       */
      dataCy(value: string): Chainable<JQuery<HTMLElement>>
    }
  }
}
